src/Exceptionless.AppHost/Extensions/ElasticsearchExtensions.cs

```diff
@@ -0,0 +1,121 @@
+            .WithEndpoint(targetPort: ElasticsearchInternalPort, name: ElasticsearchResource.InternalEndpointName)
+            .WithEnvironment("discovery.type", "single-node")
+            .WithEnvironment("xpack.security.enabled", "false")
+            .WithEnvironment("action.destructive_requires_name", "false")
+            .WithEnvironment("ES_JAVA_OPTS", "-Xms1g -Xmx1g")
+            .WithHealthCheck(healthCheckKey)
+            .PublishAsConnectionString();
+    }
+
+    public static IResourceBuilder<ElasticsearchResource> WithKibana(this IResourceBuilder<ElasticsearchResource> builder, Action<IResourceBuilder<KibanaResource>>? configureContainer = null, string? containerName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        if (builder.ApplicationBuilder.Resources.OfType<KibanaResource>().SingleOrDefault() is { } existingKibanaResource)
+        {
+            var builderForExistingResource = builder.ApplicationBuilder.CreateResourceBuilder(existingKibanaResource);
+            configureContainer?.Invoke(builderForExistingResource);
+            return builder;
+        }
+        else
+        {
+            containerName ??= $"{builder.Resource.Name}-kibana";
+
+            builder.ApplicationBuilder.Services.TryAddLifecycleHook<KibanaConfigWriterHook>();
+
+            var resource = new KibanaResource(containerName);
+            var resourceBuilder = builder.ApplicationBuilder.AddResource(resource)
+                                      .WithImage(ElasticsearchContainerImageTags.KibanaImage, ElasticsearchContainerImageTags.Tag)
+                                      .WithImageRegistry(ElasticsearchContainerImageTags.Registry)
+                                      .WithHttpEndpoint(targetPort: KibanaPort, name: containerName)
+                                      .WithEnvironment("xpack.security.enabled", "false")
+                                      .ExcludeFromManifest();
+
+            configureContainer?.Invoke(resourceBuilder);
+
+            return builder;
+        }
+    }
+
+    public static IResourceBuilder<ElasticsearchResource> WithDataVolume(this IResourceBuilder<ElasticsearchResource> builder, string? name = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        return builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/usr/share/elasticsearch/data");
+    }
+
+    public static IResourceBuilder<ElasticsearchResource> WithDataBindMount(this IResourceBuilder<ElasticsearchResource> builder, string source)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(source);
+
+        return builder.WithBindMount(source, "/usr/share/elasticsearch/data");
+    }
+}
+
+internal static class ElasticsearchContainerImageTags
+{
+    public const string Registry = "docker.elastic.co";
+    public const string Image = "elasticsearch/elasticsearch";
```

Comment:
How will this work with mapper-size plugin? Guessing that is a production only toggle.

Reply:
Fixed